samples/luis-bot-es6/app.js

```diff
@@ -49,90 +49,91 @@ adapter.use(state);
+                    // Resolve entities returned from LUIS, and save these to state
+                    let title = state.title = res.entities['Calendar.Subject'];
+                    let date = res.entities.builtin_datetimeV2_date;
+                    let time = res.entities.builtin_datetimeV2_time;
+                    let datetime = res.entities.builtin_datetimeV2_datetime;
+                    // TODO: The datetime parsing below should be
+                    // updated to use TIMEX parsing instead of moment parsing
+                    if (datetime) {
+                        let dtvalue = datetime[0];
+                        state.date = moment(dtvalue).format('DD-MM-YYYY');
+                        state.time = moment(dtvalue).format('hh:mm a');
+                    } else if (date) {
+                        let datevalue = date[0];
+                        state.date = moment(datevalue).format('DD-MM-YYYY');
+                    } else if (time) {
+                        let timevalue = time[0];
+                        // Assume user is talking about today
+                        let datetime = moment(timevalue, 'hh:mm a');
+                        state.date = datetime.format('DD-MM-YYYY');
+                        state.time = datetime.format('hh:mm a');
+                    }
+
+                    let reminder = (state.reminder = {
+                        title: title ? title.entity : null,
+                        date: state.date ? state.date : null,
+                        time: state.time ? state.time : null
+                    });
+
+                    return dc.begin('addReminder');
+
+                    // Start deleteReminder dialog
+                } else if (topIntent === 'Calendar_Delete') {
+                    return dc.begin('deleteReminder');
+
+                    // Start showReminders
+                } else if (topIntent === 'Calendar_Find') {
+                    return dc.begin('showReminders');
+
+                    // Check for cancel
+                } else if (utterance === 'cancel') {
+                    if (dc.activeDialog) {
+                        const res = await context.sendActivity(`Ok... Cancelled.`)
+                        dc.endAll();
                     } else {
-                        return dc.continue().then(res => {
-                            // Return default message if nothing replied.
-                            if (!context.responded) {
-                                await context.sendActivity(
-                                    `Hi! I'm a simple reminder bot. I can help add reminders, delete and show them.`
-                                );
-                            }
-                        });
+                        await context.sendActivity(`Nothing to cancel.`);
+                    }
+
+                    // Continue current dialog
+                } else {
+                    const res = await dc.continue()
```

Comment:
>const res = [](start = 20, length = 11)

Why are you capturing this? You don't do anything with it and honestly its not needed.